cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -211,24 +213,19 @@ public static void validateConfig(final SecureStore secureStore,
   }
 
   /**
-   * Commits and pushes the changes of a given file under the repository root
-   * path.
+   * Commits and pushes the changes of given files under the repository root path.
+   * Processes the hashes to a list of given type S
    *
-   * @param commitMeta  Details for the commit including author, committer and
-   *                    commit message
-   * @param fileChanged The relative path to repository root where the file is
-   *                    updated
-   * @return the hash of the written file.
-   * @throws GitAPIException          when the underlying git commands fail
-   * @throws NoChangesToPushException when there's no file changes for the
-   *                                  commit
-   * @throws GitOperationException    when failed to get filehash due to IOException
-   *                                  or the {@link PushResult} status is not OK
-   * @throws SourceControlException   when failed to get the fileHash before
-   *                                  push
+   * @param commitMeta Details for the commit including author, committer and commit message
+   * @param filesChanged Supplies relative paths to repository root of the updated files
+   * @throws GitAPIException when the underlying git commands fail
+   * @throws NoChangesToPushException when there's no file changes for the commit
+   * @throws GitOperationException when failed to get filehash due to IOException or the
+   *     {@link PushResult} status is not OK
+   * @throws SourceControlException when failed to get the fileHash before push
    */
-  public String commitAndPush(final CommitMeta commitMeta,
-      final Path fileChanged)
+  public <T, S extends Supplier<Path>> Collection<T> commitAndPush(CommitMeta commitMeta, Collection<S> filesChanged,
+      BiFunction<S, String, T> hashConsumer)
```

Comment:
Missing doc for `hashConsumer` and the @return

Reply:
Updated the javadoc